.jenkinsci/docker-pull-or-build.groovy

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env groovy
+
+def remoteFilesDiffer(f1, f2) {
+  sh "curl -L -o /tmp/${env.GIT_COMMIT}/f1 --create-dirs ${f1}"
+  sh "curl -L -o /tmp/${env.GIT_COMMIT}/f2 ${f2}"
+  diffExitCode = sh(script: "diff -q /tmp/${env.GIT_COMMIT}/f1 /tmp/${env.GIT_COMMIT}/f2", returnStatus: true)
+  if (diffExitCode == 0) {
+    return false
+  }
+  return true
+}
+
+def dockerPullOrUpdate() {
+  def platform = sh(script: 'uname -m', returnStdout: true).trim()
+  def commit = sh(script: "echo ${BRANCH_NAME} | md5sum | cut -c 1-8", returnStdout: true).trim()
+  if (remoteFilesDiffer("https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_COMMIT}/docker/develop/${platform}/Dockerfile", "https://raw.githubusercontent.com/hyperledger/iroha/${env.GIT_PREVIOUS_COMMIT}/docker/develop/${platform}/Dockerfile")) {
```

Comment:
It seems that number of conditional branches can be reduced, e.g. now it is 
```
if (differ in branch)
  if (develop)
    build image; push
  else
    build image
```
can be simplified to
```
if (differ in branch)
  build image
  if (develop)
    push
```